### PR TITLE
Assign Discord role endpoint

### PR DESCRIPTION
**Pull request by ryanlyrl**
Adds new endpoint `/user/addRoles` which assigns the specified roles in the .env to the currently logged in user.

### CODE REVIEW

File: src/controllers/userController.ts [hunk 1]
```diff
@@ -7,11 +7,15 @@ import { v4 as uuidv4 } from 'uuid';
...
 import * as email from '../middleware/email';
 import {EmailConfirmToken} from '../entity/EmailConfirmToken';
+import axios, {Method} from 'axios';
 
 const HOSTNAME = process.env.EXTERNAL_HOSTNAME;
 const CONFIRM_LINK = process.env.CONFIRM_LINK || '';
 const CONFIRM_TEMPLATE = process.env.CONFIRM_TEMPLATE || '';
 
+const DISCORD_URL = process.env.DISCORD_URL;
+const DISCORD_ROLES = process.env.DISCORD_ROLES;
```
Comment (dellisd) [inline src/controllers/userController.ts]: The naming of this variable is weird. We're only setting a _single_ role, right? Unless this is supposed to be a list that gets passed in somehow?

**Review by dellisd — COMMENTED** [review]
I suppose this will be a useful endpoint to have so we can manually set roles, but I think for the purposes of the event the event role ("2021") should be applied once they've linked their discord accounts. The frontend wouldn't be making a second api call to assign the role, unless I'm misunderstanding the purpose of this PR?

**Comment by ryanlyrl** [comment]
This branch was a bit old, so our requirements might have changed now... But I think we wanted people to accept the MLH terms or something before we have them a role? Which means it can't happen on sign up.

**Comment by dellisd** [comment]
That's a good point. I suppose it's theoretically possible to handle that though. If they registered via discord, only apply the role after registration? And otherwise, only apply when they link their discord account (which is after registration anyway)?

**Review by dellisd — APPROVED** [review]

**Pull request merged**